modules/lx-autoscale/watchmaker-lx-autoscale.template.cfn.json

```diff
@@ -1708,10 +1708,10 @@
                                 "yum -y install gcc python-devel libffi-devel openssl-devel\n",
                                 "pip install",
                                 " --index-url=\"$PYPI_URL\"",
-                                " --upgrade cffi\n",
+                                " --upgrade 'pycparser<2.19;python_version<\"2.7\"' cffi\n",
                                 "pip install",
                                 " --index-url=\"$PYPI_URL\"",
-                                " --upgrade 'cryptography<2.2;python_version<\"2.7\"' 'cryptography;python_version>=\"2.7\"'",
+                                " --upgrade 'pycparser<2.19;python_version<\"2.7\"' 'cryptography<2.2;python_version<\"2.7\"' 'cryptography;python_version>=\"2.7\"'",
```

Comment:
Is `pycparser<2.19;python_version<\"2.7\"` necessary in the second `pip install`?

Reply:
I tried without first.

Unfortunately, it is required, yes. The `--upgrade` tells pip to upgrade the target package if it is already installed, *AND* also upgrade all the dependencies of the targeted package. Generally this is what we want here, because cryptography and its deps are subject to security updates. However, since pycparser is a dependency of cryptography, it attempts to upgrade it to latest (2.19) which then fails.

Reply:
Thanks for the explanation.  I see why it's needed.